src/main/java/seedu/agendum/model/Model.java

```diff
@@ -22,7 +24,21 @@
 
     /** Adds the given task */
     void addTask(Task task) throws UniqueTaskList.DuplicateTaskException;
-
+    
+    /** Rename the given task */
+    void renameTask(ReadOnlyTask target, Name newTaskName)
+            throws UniqueTaskList.TaskNotFoundException, UniqueTaskList.DuplicateTaskException;
+    
+    /**Schedules the given task */
+    void scheduleTask(ReadOnlyTask target, LocalDateTime startDateTime, LocalDateTime endDateTime)
```

Comment:
Do you all agree with the following format?
Time can be null?
Alternatively, it is possible to clone the task in the Logic/respective command